doc/source/serve/production.md

```diff
@@ -567,3 +557,216 @@ This section summarizes the best practices when deploying to production:
+kubernetes                                         ClusterIP   10.100.0.1       <none>        443/TCP                                          62d
+# A head node service maintained by the RayService.
+rayservice-sample-head-svc                         ClusterIP   10.100.34.24     <none>        6379/TCP,8265/TCP,10001/TCP,8000/TCP,52365/TCP   24m
+# A dashboard agent service maintained by the RayCluster.
+rayservice-sample-raycluster-qd2vl-dashboard-svc   ClusterIP   10.100.109.177   <none>        52365/TCP                                        24m
+# A head node service maintained by the RayCluster.
+rayservice-sample-raycluster-qd2vl-head-svc        ClusterIP   10.100.180.221   <none>        6379/TCP,8265/TCP,10001/TCP,8000/TCP,52365/TCP   24m
+# A serve service maintained by the RayService.
+rayservice-sample-serve-svc                        ClusterIP   10.100.39.92     <none>        8000/TCP                                         24m
+```
+
+### Querying the application
+
+Once the `RayService` is running, we can query it over HTTP using the service created by the controller.
+This service can be queried directly from inside the cluster, but to access it from your laptop you'll need to configure a Kubernetes ingress or use port forwarding as below:
+
+```console
+$ kubectl port-forward service/rayservice-sample-serve-svc 8000
+$ curl -X POST -H 'Content-Type: application/json' localhost:8000 -d '["MANGO", 2]'
+6
+```
+
+### Getting the status of the application
+
+As the `RayService` is running, the `KubeRay` controller will continually health check and monitor it and write status updates to the CR.
+You can view the status of the application using `kubectl describe`.
+This will include the status of the cluster, any events that happened such as health check failures or restarts, and the same application-level statuses reported by [`serve status`](serve-in-production-inspecting).
+
+```console
+$ kubectl get rayservices
+NAME                AGE
+rayservice-sample   7s
+
+$ kubectl describe rayservice rayservice-sample
+```
+
+### Updating the application
+
+To update the `RayService`, modify the manifest and apply it use `kubectl apply`.
+There are two types of updates that can occur:
+- *Application-level updates*: when only the Serve config options are changed, the update will be applied _in-place_ on the same Ray cluster. This enables lightweight updates such as scaling a deployment up or down or modifying autoscaling parameters.
+- *Cluster-level updates*: when the `RayCluster` config options are changed, such as updating the container image for the cluster, it may result in a cluster-level update. In this case, a new cluster will be started, the application will be deployed to it, and once it is running the Kubernetes service will be updated to point at the new cluster. There should not be any downtime for the application.
```

Comment:
Should we clarify that this requires enough resources to duplicate the running cluster? Or is that a natural assumption?